incognia.go

```diff
@@ -298,12 +312,17 @@ func (c *Client) doRequest(request *http.Request, response interface{}) error {
 }
 
 func (c *Client) authorizeRequest(request *http.Request) error {
-	token, err := c.tokenManager.getToken()
+	token, err := c.tokenProvider.GetToken()
 	if err != nil {
 		return err
 	}
 
-	request.Header.Add("Authorization", fmt.Sprintf("%s %s", token.TokenType, token.AccessToken))
+	accessToken, ok := token.(*accessToken)
+	if !ok {
+		return ErrExpectingOAuthToken
+	}
+
+	request.Header.Add("Authorization", fmt.Sprintf("%s %s", accessToken.TokenType, accessToken.AccessToken))
```

Comment:
Instead of forcing a specific token type, you can do like Go's OAuth2 library an let the token implementation decide how to authenticate the request:

https://github.com/golang/oauth2/blob/master/token.go#L74-L81